commands/doit.go

```diff
@@ -51,36 +53,59 @@ var Writer = os.Stdout
 // Trace toggles http tracing output.
 var Trace bool
 
+// cfgFile is the location of the config file
+var cfgFile string
+
+var cfgFileName = ".doctlcfg"
+
 func init() {
-	viper.SetConfigType("yaml")
+	cobra.OnInitialize(initConfig)
 
-	DoitCmd.PersistentFlags().StringVarP(&Token, "access-token", "t", "", "DigitalOcean API V2 Access Token")
+	DoitCmd.PersistentFlags().StringVarP(&cfgFile, "config", "c", "", "config file (default is $HOME/.doctlcfg)")
+	DoitCmd.PersistentFlags().StringVarP(&Token, "access-token", "t", "", "API V2 Access Token")
 	DoitCmd.PersistentFlags().StringVarP(&Output, "output", "o", "text", "output formt [text|json]")
 	DoitCmd.PersistentFlags().BoolVarP(&Verbose, "verbose", "v", false, "verbose output")
-	DoitCmd.PersistentFlags().BoolVarP(&Trace, "trace", "", false, "verbose output")
+	DoitCmd.PersistentFlags().BoolVarP(&Trace, "trace", "", true, "trace api access")
+
+	viper.SetEnvPrefix("DIGITALOCEAN")
+	viper.BindEnv("access-token", "DIGITALOCEAN_ACCESS_TOKEN")
+	viper.BindPFlag("access-token", DoitCmd.PersistentFlags().Lookup("access-token"))
+	viper.BindPFlag("output", DoitCmd.PersistentFlags().Lookup("output"))
+	viper.BindEnv("enable-beta", "DIGITALOCEAN_ENABLE_BETA")
+
+	addCommands()
 }
 
-// LoadConfig loads out configuration.
-func LoadConfig() error {
-	cf, err := doctl.NewConfigFile()
-	if err != nil {
-		return err
+func initConfig() {
+	if cfgFile == "" {
+		cfgFile = filepath.Join(homeDir(), ".doctlcfg")
 	}
 
-	r, err := cf.Open()
-	if err != nil {
-		return fmt.Errorf("can't open configuration file: %v", err)
+	viper.SetConfigType("yaml")
+	viper.SetConfigFile(cfgFile)
+
+	viper.AutomaticEnv()
+
+	if _, err := os.Stat(cfgFile); err == nil {
+		if err := viper.ReadInConfig(); err != nil {
+			log.Fatalln("reading initialization failed:", err)
+		}
 	}
 
-	return viper.ReadConfig(r)
+	viper.SetDefault("output", "text")
+
 }
 
-// Init initializes the root command.
-func Init() *Command {
-	initializeConfig()
-	addCommands()
+func homeDir() string {
+	return os.Getenv("PATH")
```

Comment:
should be `HOME`? `PATH` is a colon separated string of path where binaries reside, not what you want I think